consensus/src/consensus_observer/observer.rs

```diff
@@ -640,28 +640,37 @@ impl ConsensusObserver {
 
         // TODO: verify the ordered block!
 
-        // If the block is a child of our last block, we can insert it
-        if self.get_last_block().id() == blocks.first().unwrap().parent_id() {
-            // Insert the ordered block into the pending blocks
-            self.pending_ordered_blocks
-                .insert_ordered_block(ordered_block);
-
-            // If we are not in sync mode, forward the blocks to the execution pipeline
-            if self.sync_handle.is_none() {
-                debug!(
+        if let Some(block) = blocks.first() {
```

Comment:
we already verify blocks is not empty right above, I think this is the right place to do unwrap/expect